server/src/main/java/org/candlepin/model/ConsumerCurator.java

```diff
@@ -560,6 +560,21 @@ public void updateLastCheckin(Consumer consumer, Date checkinDate) {
             .executeUpdate();
     }
 
+    @Transactional
+    public void heartbeatUpdate(final String reporterId, final Date checkIn) {
+        final String hql = "" +
+            "UPDATE cp_consumer" +
+            " SET lastcheckin = :checkin" +
+            " WHERE id IN (SELECT consumer_id" +
+            " FROM cp_consumer_hypervisor" +
+            " WHERE reporter_id = :reporter)";
```

Comment:
This needs to be scoped to the current organization.

Reply:
Don't need a subselect:

            "UPDATE cp_consumer a, cp_consumer_hypervisor b, cp_owner c" +
            " SET a.lastcheckin = :checkin" +
            " WHERE a.id = b.consumer_id" +
            " AND b.reporter_id = :reporter" +
            " AND a.owner_id = c.id" +
            " AND c.account = :orgKey;"

Reply:
Done.